### PR TITLE
cli: Add commands to print useful information from store data files.

### PR DESCRIPTION
**Pull request by bdarnell**
Motivated by investigations in #3896.

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/4123)

### CODE REVIEW

**Comment by petermattis** [comment]
LGTM

---

Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[cli/debug.go, line 125 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4123#-K9dwsRcVOXxMIPNfH5v:-K9dwsRdb-k7-EhSUZtB:1618986795) ([raw file](https://github.com/cockroachdb/cockroach/blob/e5d3b0a244118988abb5da3ff02e8aeed267cda6/cli/debug.go#L125)):*
#3529 talks about adding something like an `-X` flag (if you remember the `bt` command) which would hide commands like these. Not sure if that is worth it or if this `debug` section is sufficient. 

@cuongdo We should unify the `kv` commands with whatever is decided here. 

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4123#-:-K9dxN0jhZy1HayqJnM1:676834233)*

File: cli/debug.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+func printKey(kv engine.MVCCKeyValue) (bool, error) {
+	fmt.Printf("%q\n", kv.Key)
+
+	return false, nil
+}
+
+func runDebugKeys(cmd *cobra.Command, args []string) error {
+	stopper := stop.NewStopper()
+	//defer stopper.Stop()
```
Comment (mberhault) [inline cli/debug.go]: why is this commented? currently broken, or unintentional?

**Comment by mberhault** [comment]
LGTM.

The debug name should be sufficient for now. I expect whoever adds a `-X` flag will audit the commands and apply it where needed.

**Comment by cuongdo** [comment]
@mberhault Yes, I'll add the `-X` flag where appropriate.

File: cli/debug.go (hunk 1 above)
Comment (bdarnell) [inline cli/debug.go]: Uncommented it. I was trying to debug a panic and the stopper was clobbering the stack trace.

**Comment by bdarnell** [comment]
Is the `bt` tool the only precedent for the `-X` flag or is it a more common convention? It looks like it would be a pain to support in the cobra package. I think that a prefix command like `debug` is enough of a warning.

**Pull request merged**